crates/typst/src/export/pdf/gradient.rs

```diff
@@ -301,3 +356,186 @@ fn transform_to_array(ts: Transform) -> [f32; 6] {
         ts.ty.to_f32(),
     ]
 }
+
+/// Low level representation of a Coons Patch in PDF.
+#[repr(C, packed)]
+#[derive(Clone, Copy)]
+struct Patch {
+    flag: u8,
+    points: [[u16; 2]; 12],
+    colors: [[u16; 3]; 4],
+}
+
+unsafe impl bytemuck::Pod for Patch {}
+unsafe impl bytemuck::Zeroable for Patch {}
```

Comment:
I think it would be safer to not have this type and instead either use bytemuck::cast_slice on the individual fields or just construct the byte array by hand. Especially since you still need the whole `to_be`.